### ISSUE TITLE
Pytorch Should have a normal_like operator

### ISSUE THREAD

**Issue by eellison** [pytorch/pytorch#52334]
## 🚀 Feature: normal_like operator

`aten::normal_(Tensor(a!) self, float mean=0, float std=1, *, Generator? generator=None) -> Tensor(a!) ` does not have a good functionally equivalent op, which makes it hard for backends with limited representation & optimizers. 

Other similar ops - `fill_, randn_, zero_,` all have `_like` operators. The only way to replicate it `normal_` without mutation is the `normal(float mean, float std, int[] size, *, Generator? generator=None, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor` schema, which requires inserting seven operators into graph, see [here](https://github.com/pytorch/pytorch/pull/52175/files#diff-c098125f9cb12f6523dba61c6eebd88ba0f49914261bc3879b83526c81b49991R45). For consistency with other _like operators, and ease of use for other backends a normal_like operator would be useful.

**Comment by rgommers** [comment]
Looking at some other functions:
- `rand`, `rand_like`
- `randn`, `randn_like`
- `randint`, `randint_like`
- `randperm` --> no `_like` equivalent

For `torch.normal` (as well as for `randperm`) I'm not sure a `_like` function makes sense. The `_like` normally indicates "give me a tensor with the same shape and dtype as that other tensor". So your original function should be able to generate arbitrary-shape tensors. `torch.normal` cannot do this, it doesn't have a `size` or `shape` parameter.

The same applies to `randperm`, no size/shape parameter.

**Comment by rgommers** [comment]
Of course, we can adapt `torch.normal` to have a `size` argument. That would make it more consistent with `np.random.normal` as well.

**Comment by mruberry** [comment]
Normal actually does have a shape parameter on one of its many signatures:

https://github.com/pytorch/pytorch/blob/0c455332e8ca7b3c426b7eb95b696eef7bedee47/aten/src/ATen/native/native_functions.yaml#L6616

Interestingly the inplace version of the operation only implements the float x float variant, which is a UX gap.

normal_like() seems like a fine addition. It'd be nice to expand the inplace version of the operation to also support tensors or floats while we did it.

**Comment by eellison** [comment]
thanks for the discussion ! The specific schema i'm proposing, which would clean up the functionalization pass a bit,  is `aten::normal_like(Tensor(a) self, float mean=0, float std=1, *, Generator? generator=None)`, but i agree that there might be other follow up schemas to round out consistency
